shell_turtlestein.py

```diff
@@ -113,6 +115,8 @@ def run(self):
                                                    on_done, None, None)
         for (setting, value) in list(settings().get('input_widget').items()):
             inputview.settings().set(setting, value)
+        if run_previous_command(args, self.cmd_history[-1]):
```

Comment:
Can we move this condition up so that we can skip showing the input panel and doing  `inputview` setup when we're just running the previous command?